ui/components/SimpleDialog.vue

```diff
@@ -0,0 +1,311 @@
+<template>
+  <Teleport to=".dialogs-container">
+    <div
+      v-if="value"
+      v-bind="$attrs"
+      class="dialog"
+      :style="{ height: containerHeight }"
+    >
+      <div
+        class="dialog__container"
+        :style="{ height, width }"
+      >
+        <div class="dialog__header">
+          <slot name="header">
+            <p class="dialog__title">{{ title }}</p>
+          </slot>
+        </div>
+        <div class="dialog__body">
+          <div
+            v-if="$slots.sidebar"
+            class="dialog__sidebar"
+          >
+            <slot name="sidebar" />
+          </div>
+          <div class="dialog__content-wrapper">
+            <div class="dialog__content">
+              <slot name="default" />
+            </div>
+            <div class="dialog__actions">
+              <template v-for="action in computedActions">
+                <div
+                  v-if="action.key === 'spacer'"
+                  :key="action.key"
+                  class="dialog__spacer"
+                />
+                <ui-button
+                  v-else
+                  :key="action.key"
+                  class="dialog__action"
+                  :disabled="action.disabled"
+                  :color="action.color || COLORS_DICT.TEXT"
+                  height="36px"
+                  background-color="transparent"
+                  @clicked="action.handler"
+                >
+                  <ui-icon
+                    v-if="action.loading"
+                    class="dialog__action-icon"
+                    :color="action.color || COLORS_DICT.TEXT"
+                    icon-name="connectLoaderAnimated"
+                    size="24"
+                  />
+                  <span
+                    v-else
+                    class="dialog__action-label"
+                  >
+                    {{ action.label }}
+                  </span>
+                </ui-button>
+              </template>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </Teleport>
+</template>
+
+<script setup>
+import { inject, watch, onBeforeUnmount, computed, ref } from 'vue';
+import { useRouter } from 'vue-router';
+
+import { COLORS_DICT } from '~/constants/colors';
+
+const value = defineModel({
+  type: Boolean,
+  default: false,
+});
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  actions: {
+    type: Array,
+    required: true,
+    validator(value) {
+      return value.every((action) =>
+        ['cancel', 'close', 'details', 'save', 'submit', 'spacer', 'next', 'back'].includes(action),
```

Comment:
this can be a const array

Reply:
Same as one of the comments above, compiler macros can't use external variables.